extensions/standard-processors/processors/SplitText.cpp

```
@@ -0,0 +1,378 @@
+  ++emitted_fragment_index_;
+}
+
+void ReadCallback::createHeaderOnlyFragmentFlow(const SplitTextFragmentGenerator::Fragment& header_fragment) {
+  gsl_Expects(split_text_config_.remove_trailing_new_lines);  // This is only possible if the split fragment has no content and the endlines are trimmed
+  auto header_only_flow = session_.clone(flow_file_, gsl::narrow<int64_t>(header_fragment.fragment_offset), gsl::narrow<int64_t>(header_fragment.fragment_size - header_fragment.endline_size));
+  if (!header_only_flow) {
+    logger_->log_error("Failed to clone header only fragment flow!");
+    return;
+  }
+  logger_->log_debug("Creating a header only fragment with fragment index: {} fragment size: {}", emitted_fragment_index_, header_only_flow->getSize());
+  setAttributesOfDoneSegment(*header_only_flow, 0);
+  results.push_back(header_only_flow);
+}
+
+void ReadCallback::mergeHeaderAndFragmentFlows(const std::shared_ptr<core::FlowFile>& header_flow, const SplitTextFragmentGenerator::Fragment& fragment, size_t fragment_trim_size) {
+  auto fragment_flow = session_.clone(flow_file_, gsl::narrow<int64_t>(fragment.fragment_offset), gsl::narrow<int64_t>(fragment.fragment_size - fragment_trim_size));
+  if (!fragment_flow) {
+    logger_->log_error("Failed to clone fragment flow!");
+    return;
+  }
+  auto merged_flow = session_.clone(header_flow);  // clone header to copy attributes
+  if (!merged_flow) {
+    logger_->log_error("Failed to clone merged fragment flow!");
+    return;
+  }
+  session_.write(merged_flow, [this, &fragment_flow, &header_flow](const std::shared_ptr<io::OutputStream>& output_stream) -> int64_t {
+    session_.read(header_flow, [&output_stream](const std::shared_ptr<io::InputStream>& header_input_stream) -> int64_t {
+      return internal::pipe(*header_input_stream, *output_stream);
+    });
```

Any errors are ignored here, is this intentional? If so, could you add a comment explaining why?